### PR TITLE
Store matrices in COO format

### PR DESCRIPTION
**Pull request by mtanneau**
Addresses the second items in #11.

Specifically, instead of storing matrices `Q` and `A`as `SparseMatrixCSC`, they are stored in COO format (and can converted to `SparseMatrixCSC` later on).
AFAIK, the only affected package would be [QuadraticModels.jl](https://github.com/JuliaSmoothOptimizers/QuadraticModels.jl) which uses the `QPSData` structure. Easy to fix though.

Thus,
```julia
    Q::SparseMatrixCSC{Float64,Int}  # quadratic term in objective
    A::SparseMatrixCSC{Float64,Int}  # constraint matrix
```
becomes
```julia
    # Quadratic objective, in COO format
    qrows::Vector{Int}
    qcols::Vector{Int}
    qvals::Vector{Int}
    # Constraint matrix, in COO format
    arows::Vector{Int}
    acols::Vector{Int}
    avals::Vector{Int}
```

Consequently, I removed `SparseArrays` from the dependencies (it is now an extra dependency for the tests).

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
-SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
-
 [compat]
 julia = "^1.0.0"
 
 [extras]
+SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
+
```
Comment (dpo) [inline Project.toml]: Extra blank line.

File: src/readqps.jl [hunk 2]
```diff
@@ -15,8 +15,14 @@ mutable struct QPSData
     objsense::Symbol                 # :min, :max or :notset
     c0::Float64                      # constant term in objective
     c::Vector{Float64}               # linear term in objective
-    Q::SparseMatrixCSC{Float64,Int}  # quadratic term in objective
-    A::SparseMatrixCSC{Float64,Int}  # constraint matrix
+    # Quadratic objective, in COO format
+    qrows::Vector{Int}
+    qcols::Vector{Int}
+    qvals::Vector{Int}
```
Comment (dpo) [inline src/readqps.jl]: `qvals` should have eltype `Float64`.

File: src/readqps.jl [hunk 3]
```diff
@@ -15,8 +15,14 @@ mutable struct QPSData
...
+    # Constraint matrix, in COO format
+    arows::Vector{Int}
+    acols::Vector{Int}
+    avals::Vector{Int}
```
Comment (dpo) [inline src/readqps.jl]: Float64

File: src/readqps.jl [hunk 4]
```diff
@@ -432,7 +438,7 @@ function read_quadobj_section(qps, varnames)
         pos = position(qps)
     end
     seek(qps, pos)  # backtrack to beginning of line
-    sparse(qrows, qcols, qvals, nvar, nvar)
+    return qrows, qcols, qvals
```
Comment (dpo) [inline src/readqps.jl]: We didn't use `return` at other places. Either add it everywhere or leave it out.

File: src/readqps.jl [hunk 5]
```diff
@@ -312,6 +318,7 @@ function read_ranges_section!(qps, connames, contypes, lcon, ucon)
         pos = position(qps)
     end
     seek(qps, pos)  # backtrack to beginning of line
+    return nothing
```
Comment (mtanneau) [inline src/readqps.jl]: The return value didn't seem to be used, so I added a `return nothing` here

**Comment by dpo** [comment]
Thank you.

**Comment by mtanneau** [comment]
> Thank you.

I'm just getting started :)

**Pull request merged**